src/v/cluster/rm_stm.cc

```diff
@@ -1711,6 +1711,8 @@ ss::future<> rm_stm::apply(model::record_batch b) {
         }
     } else if (hdr.type == model::record_batch_type::tx_checkpoint) {
         apply_checkpoint(std::move(b));
+    } else if (hdr.type == model::record_batch_type::tx_checkpoint_purge) {
+        apply_checkpoint_purge(std::move(b));
```

Comment:
same thing there, let the purge reset the cache instead of tinkering with the hot data